src/server/game/Entities/Creature/Creature.cpp

```diff
@@ -3343,8 +3343,22 @@ void Creature::AtEngage(Unit* target)
 
     MovementGeneratorType const movetype = GetMotionMaster()->GetCurrentMovementGeneratorType();
     if (movetype == WAYPOINT_MOTION_TYPE || movetype == POINT_MOTION_TYPE || (IsAIEnabled() && AI()->IsEscorted()))
+    {
         SetHomePosition(GetPosition());
 
+        // if its a vehicle, set the home positon of every creature passenger at engage
+        // so that they are in combat range if hostile
+        if (Vehicle* vehicle = GetVehicleKit())
+        {
+            for (auto seat = vehicle->Seats.begin(); seat != vehicle->Seats.end(); ++seat)
+                if (Unit* passenger = vehicle->GetPassenger(seat->first))
+                    if (Creature* creature = passenger->ToCreature())
+                    {
```

Comment:
not sure what we usually do, but probably have no { } if there are no "else" cases and if there is only 1 line.
Everything else looks ok imo.

Reply:
Oh, you're right, It was a two liner before.